File: stack&queue/leetcode_682/solution.py
class Solution:
    def calPoints(self, operations: List[str]) -> int:
        stack = []

        for i in range(len(operations)):
            operation = operations[i]
        
            if operation == 'D':
                stack.append(stack[-1] * 2)
            elif operation == 'C':
                stack.pop()
            elif operation == '+':
                stack.append(stack[-1] + stack[-2])
            else:
                stack.append(int(operation))
        
        return sum(stack)

File: stack&queue/leetcode_682/solution.js
/**
 * @param {string[]} operations
 * @return {number}
 */

var calPoints = function(operations) {
    const stack = [];

    for(let i=0; i<operations.length; i++) {
        const operation = operations[i];

        switch(operation) {
            case 'D':
                stack.push(stack[stack.length - 1] * 2);
                break;

            case 'C':
                stack.pop();
                break;

            case '+':
                stack.push(stack[stack.length - 1] + stack[stack.length - 2]);
                break;
            
            default:
                stack.push(Number(operation))
        }
    }

    return stack.reduce((a, b) => (a + b), 0);
};